modules/widgets/mailchimp.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+if ( ! class_exists( 'Jetpack_MailChimp_Subscriber_Popup_Widget' ) ) {
+
+	//register MailChimp Subscriber Popup widget
+	function jetpack_mailchimp_subscriber_popup_widget_init() {
+		register_widget( 'Jetpack_MailChimp_Subscriber_Popup_Widget' );
+	}
+
+	add_action( 'widgets_init', 'jetpack_mailchimp_subscriber_popup_widget_init' );
+
+	/**
+	 * Add a MailChimp subscription form.
+	 */
+	class Jetpack_MailChimp_Subscriber_Popup_Widget extends WP_Widget {
+
+		/**
+		 * Constructor
+		 */
+		function __construct() {
+			parent::__construct(
+				'widget_mailchimp_subscriber_popup',
+				/** This filter is documented in modules/widgets/facebook-likebox.php */
+				apply_filters( 'jetpack_widget_name', __( 'MailChimp Subscriber Popup', 'jetpack' ) ),
+				array(
+					'classname'                   => 'widget_mailchimp_subscriber_popup',
+					'description'                 => __( 'Allows displaying a popup subscription form to visitors.', 'jetpack' ),
+					'customize_selective_refresh' => true,
+				)
+			);
+		}
+
+		/**
+		 * Outputs the HTML for this widget.
+		 *
+		 * @param array $args     An array of standard parameters for widgets in this theme
+		 * @param array $instance An array of settings for this widget instance
+		 *
+		 * @return void Echoes it's output
+		 **/
+		function widget( $args, $instance ) {
+			$instance = wp_parse_args( $instance, array( 'code' => '' ) );
+
+			// Regular expresion that will match maichimp shortcode.
+			$regex = "(\[mailchimp_subscriber_popup[^\]]+\])";
+
+			// Check if the shortcode exists.
+			preg_match( $regex, $instance['code'], $matches );
+
+			// Process the shortcode only, if exists.
+			if ( ! empty( $matches[0] ) ) {
+				echo do_shortcode( $matches[0] );
```

Comment:
Good catch. The check has been added, so the widget will not be registered when shortcode is missing for some reason.